### DISCUSSION TITLE
The 12 recovery words should be language dependend!

### DISCUSSION THREAD

**Discussion by wehkah** [seedvault-app/seedvault#509] [Ideas]
Hi there!

I think it would be much more convenient and even more secure if Seedvault would recognize the system language and choose the 12 words accordingly, for two reasons:

1. Anyone will remember words in their own language much easier than in a "foreign tongue".
2. Word in your own language are much less vulnerable to suffer spelling mistakes, which can lead to a useless locked backup.

**Comment by t-m-w** [comment]
Just adding some background info/references to this.

Seedvault's recovery code system uses [BIP39](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki) as noted in [README.md](https://github.com/seedvault-app/seedvault#features). 

Although the BIP39 spec considers non-English wordlists to be "[strongly discouraged](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#wordlists)", the reasoning for this appears to be based on its use with cryptographic wallets, so it may not apply so strongly here.

Regardless, the BIP39 spec goes on to link to [some non-English wordlists](https://github.com/bitcoin/bips/blob/master/bip-0039/bip-0039-wordlists.md#wordlists).

**Comment by mchar7** [comment]
I'm gonna have to disagree with this. I appreciate that people prefer to have things localized in their native languages, and it's important for apps to have UI strings translatable wherever possible. In this case however, I think ensuring that everyone is using the same predictable word list is important, regardless of locale. I don't think people are reasonably expected to memorize their 12-word list, so memorability is not really important. Spelling mistakes are also not an issue since one of the main benefits of BIP39 is the rigid word list. For example, I use Canadian English whereas the word list uses American English. I would write some possible words like "favorite" as "favo**u**rite." In reality, this does not cause confusion since I can see "favorite" in the drop-down word options. Software implementing BIP39 like Seedvault should know to mark misspelled words as incorrect instead of assuming the whole seed is invalid.

In other words, while BIP39 does perpetuate the (American) English-centricity common in computer science, I think the standardized locale is worth keeping in this case where the en-US words are central to how the software works.

**Comment by wehkah** [comment]
> In this case however, I think ensuring that everyone is using the same predictable word list is important,

I find it very strange that a "predictable" list of words is supposed to be a security enforcement. 

>  I don't think people are reasonably expected to memorize their 12-word list, so memorability is not really important.

This would mean people are forced to write their 12 words on a piece of paper. But that won't happen, because paper is commonly not in use any more, so people will either store these words in a file or they will make a screen shot and keep that picture on the same device, both of which will most probably happen on the same device that those 12 words belong to.

> In reality, this does not cause confusion since I can see "favorite" in the drop-down word options.

What "drop-down" list are you talking about? The 12 words that are generated are not choosable at all.

> ...I think the standardized locale is worth keeping in this case where the en-US words are central to how the software works.

Tell this to people who don't speak any word of English at all and people who use languages that even don't use Latin characters, like Chinese, Russian, or many, many other languages in this planet. They are forced to write words in a strange language onto a piece of paper, and later, if they need these words, they are forced to enter these words — using which keyboard??? Do you think that any one who uses a Cyrillic keyboard has access to a Latin keyboard as well?

I understand your point that internally every piece of software needs some fundamental base, some sort of least common denominator, and English is the most suited language for this purpose. I'll give you that. But we're not living in the stone age any more, every device nowadays is equipped with more computing power and memory than so called "super computers" from 20 years ago, so implementing localization in ANY software is not a technical issue any more. In these days, it is only a question of motivation.

**Comment by wehkah** [comment]
> ...since I can see "favorite" in the drop-down word options.

An idea just crossed my mind, when I re-read this sentence: what about offering a drop down list, a small set, a small list of **localized** words for each of the 12 words? That way, people could roughly create some sort of _sentence_ in their mind, that will only make sense to them, but that will be much easier to remember. These _sentences_ will be far from correct grammar, in the best they will look and sound more like Yoda's speech, but "remember them people can", which is the most important point after being a security measure.
